### PR TITLE
Gen 3: Add Deoxys Forme Clause and Deoxys Camouflage Mod

### PR DESCRIPTION
**Pull request by xInstruct**
Talked in dev room. "Do nothing clause" idea from Wob

### CODE REVIEW

File: data/mods/gen3/rulesets.ts [hunk 1]
```diff
@@ -3,6 +3,27 @@ export const Formats: {[k: string]: ModdedFormatData} = {
...
+			let deoxysType;
+			for (const set of team) {
+				if (set.species === 'Deoxys' || set.species === 'Deoxys-Attack' ||
+					set.species === 'Deoxys-Defense' || set.species === 'Deoxys-Speed') {
+					if (!deoxysType) {
+						deoxysType = set.species;
+					} else if (deoxysType && deoxysType !== set.species) {
+						return [
+							`You cannot have more than one forme of Deoxys.`,
+							`(The forme of Deoxys you get is dependant on the game you own.)`,
```
Comment (KrisXV) [inline data/mods/gen3/rulesets.ts:21]:
```suggestion
							`(The forme of Deoxys you get is dependent on the game you own.)`,
```

File: data/mods/gen3/rulesets.ts [hunk 2]
```diff
@@ -3,6 +3,27 @@ export const Formats: {[k: string]: ModdedFormatData} = {
...
+		ruleset: ['Obtainable', 'Sleep Clause Mod', 'Switch Priority Clause Mod', 'Species Clause', 'Nickname Clause', 'OHKO Clause', 'Evasion Moves Clause', 'Endless Battle Clause', 'HP Percentage Mod', 'Cancel Mod', 'Deoxys Camouflage Mod'],
+	},
+	obtainable: {
+		inherit: true,
+		// Mostly hardcoded in team-validator.ts
+		onValidateTeam(team, format) {
+			let deoxysType;
+			for (const set of team) {
+				if (set.species === 'Deoxys' || set.species === 'Deoxys-Attack' ||
+					set.species === 'Deoxys-Defense' || set.species === 'Deoxys-Speed') {
```
Comment (KrisXV) [inline data/mods/gen3/rulesets.ts:15]:
```suggestion
				if (species.species.startsWith("Deoxys")) {
```

**Comment by DaWoblefet** [comment]
Summarizing notes in dev room: Deoxys formes don't seem to be obtainable outside of their home game except via ROM hacks. The Emerald disassembly, for example, only contains the base stats of regular Deoxys and Speed Deoxys, but not Attack or Defense (let alone sprites). Couldn't get non-native Deoxys formes to be saved in PKHeX either.

**Comment by HoeenCoder** [comment]
Requested as per @xInstruct's request

File: sim/team-validator.ts [hunk 3]
```diff
@@ -275,6 +276,16 @@ export class TeamValidator {
 					problems.push(`You can only have one of Pikachu-Starter or Eevee-Starter on a team.`);
 				}
 			}
+			if (dex.gen === 3 && set.species.startsWith("Deoxys")) {
+				if (!deoxysType) {
+					deoxysType = set.species;
+				} else if (deoxysType && deoxysType !== set.species) {
```
[the remark this answers is not in the document]
Comment (xInstruct) [reply sim/team-validator.ts:282]: good point!

**Comment by Marty-D** [comment]
Some thoughts: this should be a Clause because it's an agreement between players, and it shouldn't show up anywhere but Gen 3 Ubers or we're implying Deoxys is legal in every format that uses Standard. I'll add some tests.

**Comment by xInstruct** [comment]
> Some thoughts: this should be a Clause because it's an agreement between players, and it shouldn't show up anywhere but Gen 3 Ubers or we're implying Deoxys is legal in every format that uses Standard. I'll add some tests.

I'll modify it so it is not in standard but in the Gen 3 ubers format

**Comment by Zarel** [comment]
Looks good to me; thanks!

**Comment by xInstruct** [comment]
I just realized I never changed the name in the PR from Mod to Clause.

Shouldn't make a big difference but :x

**Pull request merged**